.github/workflows/ci.yml

```diff
@@ -0,0 +1,63 @@
+name: ci
+
+on: [push]
+
+env:
+  GO_VERSION: 1.24.1
+  RETENTION-DAYS: 1
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: build
+        run: cd ./cli/alphabill && go build -o ../../build/alphabill
+      - name: upload build artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: build
+          path: build
+          retention-days: ${{ env.RETENTION-DAYS }}
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: vet
+        run: go vet ./...
+      - name: test
+        run: go test ./... -coverpkg=./... -count=1 -coverprofile test-coverage.out
+      - name: upload test coverage
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-coverage
+          path: test-coverage.out
+          retention-days: ${{ env.RETENTION-DAYS }}
+
+  analyze:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Install gosec
+        run: go install github.com/securego/gosec/v2/cmd/gosec@latest
+      - name: gosec
+        run: gosec -fmt=sonarqube -out gosec_report.json -no-fail ./...
```

Comment:
not sure if there's any point in running gosec and not failing the job. Could make analyze step a manual workflow, or remove `-no-fail`. (current report already contains a blocker)